component/src/main/java/com/celements/structEditor/DefaultStructuredDataEditorService.java

```diff
@@ -225,22 +227,27 @@ private Object getCellValue(DocumentReference cellDocRef, XWikiDocument onDoc)
     return Optional.absent();
   }
 
-  Optional<BaseObject> getCellXObject(XWikiDocument cellDoc, XWikiDocument onDoc) {
+  Optional<BaseObject> getRequestXObjectInStructEditor(XWikiDocument cellDoc, XWikiDocument onDoc) {
     Optional<BaseObject> ret = Optional.absent();
-    Optional<DocumentReference> classRef = getCellClassRef(cellDoc);
-    if (classRef.isPresent()) {
-      if (context.getRequest().isPresent()) {
-        try {
-          int objNb = Integer.parseInt(context.getRequest().get().get("objNb"));
-          ret = modelAccess.getXObject(onDoc, classRef.get(), objNb);
-        } catch (NumberFormatException nfe) {
-          LOGGER.trace("unable to parse objNb from request: {}", nfe.getMessage());
-        }
-      }
-      ret = ret.or(Optional.fromNullable(modelAccess.getXObject(onDoc, classRef.get())));
+    try {
+      int objNb = Integer.parseInt(context.getRequestParameter("objNb").or("0"));
+      ret = getXObjectInStructEditor(cellDoc, onDoc, objNb);
+    } catch (NumberFormatException nfe) {
+      LOGGER.debug("unable to parse objNb from request: {}", nfe.getMessage());
+    }
+    return ret;
+  }
+
+  Optional<BaseObject> getXObjectInStructEditor(XWikiDocument cellDoc, XWikiDocument onDoc,
```

Comment:
please make it a 'private' method or if it is a service method add it to the Role interface